--- a/utils/dialog/CMakeLists.txt
+++ b/utils/dialog/CMakeLists.txt
@@ -24,7 +24,7 @@
 )
 
 target_link_libraries(hyprland-dialog PRIVATE
-    Qt6::Widgets Qt6::QuickControls2 Qt6::WaylandClientPrivate PkgConfig::hyprutils
+    Qt6::Widgets Qt6::QuickControls2 PkgConfig::hyprutils
 )
 
 
--- a/utils/update-screen/CMakeLists.txt
+++ b/utils/update-screen/CMakeLists.txt
@@ -32,7 +32,7 @@
 )
 
 target_link_libraries(hyprland-update-screen PRIVATE
-    Qt6::Widgets Qt6::QuickControls2 Qt6::WaylandClientPrivate PkgConfig::hyprutils
+    Qt6::Widgets Qt6::QuickControls2 PkgConfig::hyprutils
 )
 
 
--- a/utils/donate-screen/CMakeLists.txt
+++ b/utils/donate-screen/CMakeLists.txt
@@ -30,7 +30,7 @@
 )
 
 target_link_libraries(hyprland-donate-screen PRIVATE
-    Qt6::Widgets Qt6::QuickControls2 Qt6::WaylandClientPrivate PkgConfig::hyprutils
+    Qt6::Widgets Qt6::QuickControls2 PkgConfig::hyprutils
 )